File: backend/app/routes/notes.py
from fastapi import APIRouter,Depends
from app.services.note import NoteService
from app.models.note import NoteCreate
from app.services.auth import get_current_user

router = APIRouter()

@router.post("/notes")
async def create_note(note: NoteCreate, current_user: dict = Depends(get_current_user)):
    return await NoteService.create_note(note, current_user)

@router.get("/notes")
async def get_notes(current_user: str = Depends(get_current_user)):
    return await NoteService.get_user_notes(current_user)

@router.get("/random_quote")
async def random_note(current_user: str = Depends(get_current_user)):
    return await NoteService.get_random_note(current_user)

File: frontend/server/src/routes/notes.js
const express = require("express");
const router = express.Router();
const { validateNote } = require("../middleware/validation");
const { validateToken } = require("../middleware/auth");
const NoteService = require("../services/noteService");
const { handleAsync } = require("../utils/errorHandler");

router.post(
  "/",
  validateToken,
  validateNote,
  handleAsync(NoteService.createNote)
);
router.get("/", validateToken, handleAsync(NoteService.getNotes));
router.get(
  "/random_quote",
  validateToken,
  handleAsync(NoteService.getRandomQuote)
);

module.exports = router;
